src/core.js

```diff
@@ -1794,6 +1835,11 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
       datamap.createMap();
     }
 
+    // The `column` property has changed - dataset may be expanded or narrowed down.
+    if (settings.data === void 0 && settings.columns !== void 0) {
```

Comment:
I didn't think about it. It probably could be in the `else if`, since in the above conditionals we have all the `loadData` calls for the `updateSettings` function.

Reply:
Changed within: https://github.com/handsontable/handsontable/pull/6547/commits/cafe897983123c30ffaac2201176d8b2c3cd13ac#diff-7eb52b366866677666470e019283c8eaL1836-L1841.